src/Toolchain.ml

```diff
@@ -247,44 +249,42 @@ let setupToolChain (kind : PackageManager.t) =
 
 let makeResources kind = kind
 
-let selectAndSave () =
+let select () =
   let open Promise.O in
   let workspaceFolders = Vscode.Workspace.workspaceFolders () in
   sandboxCandidates ~workspaceFolders >>= fun candidates ->
-  selectPackageManager candidates >>| function
-  | None -> None
-  | Some choice -> (
-    match choice.status with
-    | Error s ->
-      message `Warn "This toolchain is invalid. Error: %s" s;
-      None
-    | Ok () ->
-      let (_ : unit Promise.t) = toSettings choice.packageManager in
-      Some choice.packageManager )
-
-let getLspCommand (t : PackageManager.t) : Path.t * string array =
-  let name = "ocamllsp" in
-  match t with
-  | Opam (opam, switch) -> Opam.exec opam ~switch ~args:[| name |]
-  | Esy (esy, manifest) -> Esy.exec esy ~manifest ~args:[| name |]
-  | Global -> (Path.ofString name, [||])
+  let open Promise.Option.O in
+  selectPackageManager candidates >>= fun { status; packageManager } ->
+  match status with
+  | Error s ->
+    message `Warn "This toolchain is invalid. Error: %s" s;
+    Promise.return None
+  | Ok () -> Promise.Option.return packageManager
 
-let addLspCheckArg args = Array.append args [| "--help=plain" |]
-
-let getDuneFormatter (t : PackageManager.t) : Path.t * string array =
-  let dune = "dune" in
-  let format = "format-dune-file" in
-  match t with
-  | Opam (opam, switch) -> Opam.exec opam ~switch ~args:[| dune; format |]
-  | Esy (esy, manifest) -> Esy.exec esy ~manifest ~args:[| dune; format |]
-  | Global -> (Path.ofString dune, [| format |])
+let selectAndSave () =
+  let open Promise.Option.O in
+  select () >>| fun packageManager ->
+  let (_ : unit Promise.t) = toSettings packageManager in
```

Comment:
Why don't we wait for the save to finish here?

Reply:
I don't believe the previous version waited for save to finish either.

The function already returns a promise, so this should be an easy fix.

Reply:
I see, well it's worth fixing as it's confusing otherwise.